lighthouse-core/report/html/renderer/report-renderer.js

```diff
@@ -201,15 +201,26 @@ class ReportRenderer {
     const categories = reportSection.appendChild(this._dom.createElement('div', 'lh-categories'));
 
     for (const category of report.reportCategories) {
-      if (scoreHeader) {
-        scoreHeader.appendChild(categoryRenderer.renderScoreGauge(category));
-      }
-
       const renderer = specificCategoryRenderers[category.id] || categoryRenderer;
       categories.appendChild(renderer.render(category, report.categoryGroups));
     }
 
     if (scoreHeader) {
+      const defaultGauges = [];
+      const customGauges = [];
+      for (const category of report.reportCategories) {
+        const renderer = specificCategoryRenderers[category.id] || categoryRenderer;
+        const categoryGauge = renderer.renderScoreGauge(category);
+
+        // Group gauges that aren't default at the end of the header
+        if (renderer.renderScoreGauge === categoryRenderer.renderScoreGauge) {
```

Comment:
hmm! how does equality on a function work? seems fine though i'm curious.

Reply:
> how does equality on a function work?

it's (shared) on the root prototype, so if a renderer descends from `CategoryRenderer` but doesn't override `renderScoreGauge` it'll just be the same function object.

I was originally going to do something more complicated with verifying *which* prototype the function that will be called was defined on, but it doesn't buy anything extra, and considering this is the only special gauge for now, anything much more complicated than a specialized check directly for the pwa renderer would be compounding the silliness.